### PR TITLE
Subtler fix and slight tidy up

### LINKED ISSUE #15471: Subtle refactor issues
## Reproduction:
Related to #15453, but there are numerous issues with Subtle. 

-As stated in the previous PR, despite being a multi-line text box, you cannot format text in any way. It's been reduced down to, arguably, an even worse version than the old explorer text box or lack of formatting from enforced one-line subtlers.
-If you enter in subtler text and go to the target menu, there is a cancel option. However, the cancel option still posts the emote to your own chat instead of returning out of the full emote.

### PR DESCRIPTION
**Pull request by tf-4**
fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/15471
Well, the second half. The first half is already fixed. For the love of God please atomise issue reports into distinct issues so they can be closed when complete more easily.

Also deindents a little, and removes a bit of dead code.
It was a `TODO` added in a PR 24 months ago. That's not getting done. If you want to do it, feel free.
It was being checked for each subtle / subtler emote and did literally nothing.

## Changelog
:cl:
fix: Subtler emotes will no longer send if you cancel target selection.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/verbs/code/subtle.dm [hunk 1]
```diff
@@ -92,29 +74,30 @@
...
-		else
+		if(!subtler_emote)
+			return FALSE
+		var/list/in_view = get_hearers_in_view(1, user)
+		in_view -= GLOB.dead_mob_list
+		in_view.Remove(user)
+
+		for(var/mob/mob_in_view as anything in in_view)
+			if(!istype(mob_in_view))
+				in_view.Remove(mob_in_view)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/verbs/code/subtle.dm]: I feel like you can untype the for(var/mob/mob_in_view and just make it for(var/mob_in_view
then, make the !istype a !ismob

Comment (tf-4) [reply modular_skyrat/modules/verbs/code/subtle.dm]: What's the advantage?

File: modular_skyrat/modules/verbs/code/subtle.dm [hunk 2]
```diff
@@ -92,29 +74,30 @@
...
-					target = SUBTLE_SAME_TILE_DISTANCE
-			subtler_message = subtler_emote
-		else
+		if(!subtler_emote)
+			return FALSE
+		var/list/in_view = get_hearers_in_view(1, user)
+		in_view -= GLOB.dead_mob_list
+		in_view.Remove(user)
+
+		for(var/mob/mob_in_view as anything in in_view)
```
Comment (Zonespace27) [inline modular_skyrat/modules/verbs/code/subtle.dm]: cut the `as anything`, BYOND already optimizes range/view/hearers/etc. procs when you're searching for a base type (area/obj/turf/mob)

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/verbs/code/subtle.dm (hunk 2 above)
Comment (tf-4) [inline modular_skyrat/modules/verbs/code/subtle.dm]:
```suggestion
		for(var/mob/mob_in_view in in_view)
```

File: modular_skyrat/modules/verbs/code/subtle.dm [hunk 3]
```diff
@@ -92,29 +74,30 @@
...
+		in_view -= GLOB.dead_mob_list
+		in_view.Remove(user)
+
+		for(var/mob/mob_in_view in in_view)
+			if(!istype(mob_in_view))
+				in_view.Remove(mob_in_view)
+		var/list/targets = list(SUBTLE_ONE_TILE_TEXT, SUBTLE_SAME_TILE_TEXT) + in_view
+		target = tgui_input_list(user, "Pick a target", "Target Selection", targets)
+		if(!target)
 			return FALSE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/verbs/code/subtle.dm:94]:
```suggestion
			return FALSE

```

File: modular_skyrat/modules/verbs/code/subtle.dm [hunk 4]
```diff
@@ -92,29 +74,30 @@
...
-			target = tgui_input_list(user, "Pick a target", "Target Selection", targets)
-			switch(target)
-				if(SUBTLE_ONE_TILE_TEXT)
-					target = SUBTLE_DEFAULT_DISTANCE
-				if(SUBTLE_SAME_TILE_TEXT)
-					target = SUBTLE_SAME_TILE_DISTANCE
-			subtler_message = subtler_emote
-		else
+		if(!subtler_emote)
+			return FALSE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/verbs/code/subtle.dm:83]:
```suggestion
			return FALSE

```

File: modular_skyrat/modules/verbs/code/subtle.dm [hunk 5]
```diff
@@ -92,29 +74,30 @@
...
+		var/list/in_view = get_hearers_in_view(1, user)
+		in_view -= GLOB.dead_mob_list
+		in_view.Remove(user)
+
+		for(var/mob/mob_in_view in in_view)
+			if(!istype(mob_in_view))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/verbs/code/subtle.dm]: If you removed the `as anything`, then the `!istype()` isn't really necessary anymore, I feel like.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**